### ISSUE TITLE
Unspecified behavior of char after ending quotes

### ISSUE THREAD

**Issue by hugosenari** [nushell/nushell#13263]
### Describe the bug

Any character after ending quote makes a string with quotes ended by char

Is an unspecified behavior, making it a documentation bug or spec bug, but could be useful for something.

### How to reproduce

`"a b"c == '"a b"c'`

### Expected behavior

Have this behavior specified in [string docs](https://www.nushell.sh/book/types_of_data.html#strings)

### Configuration

| key                | value                                                             |
| ------------------ | ----------------------------------------------------------------- |
| version            | 0.93.0                                                            |
| major              | 0                                                                 |
| minor              | 93                                                                |
| patch              | 0                                                                 |
| branch             |                                                                   |
| commit_hash        |                                                                   |
| build_os           | linux-x86_64                                                      |
| build_target       | x86_64-unknown-linux-gnu                                          |
| rust_version       | rustc 1.77.2 (25ef9e3d8 2024-04-09) (built from a source tarball) |
| cargo_version      | cargo 1.77.1                                                      |
| build_time         | 1980-01-01 00:00:00 +00:00                                        |
| build_rust_channel | release                                                           |
| allocator          | mimalloc                                                          |
| features           | default, sqlite, system-clipboard, trash, which                   |
| installed_plugins  |                                                                   |

### Additional context

https://imgflip.com/i/69ov8j

**Comment by fdncred** [comment]
0.93.0 is pretty old. you should try the latest main because we just landed a PR that may help with this. https://github.com/nushell/nushell/pull/13131

**Comment by hugosenari** [comment]
> 0.93.0 is pretty old. 

| key                | value                                                             |
| ------------------ | ----------------------------------------------------------------- |
| version            | 0.95.0                                                            |

```
> 'a'b
'a'b
> 'a b'b
'a b'b
> "a b"c == '"a b"c'
true
>
```

@fdncred thank you, #13131 (nice number) was merged 18h ago
I will wait v0.96.0 then.

I was hoping for it to be a feature :wink:

**Comment by fdncred** [comment]
Right, it's recent, which is why I suggested trying the latest main branch. Here's what happens now.
```nushell
❯ 'a'b
Error: nu::parser::extra_token_after_closing_delimiter

  × Invalid characters after closing delimiter
   ╭─[entry #6:1:4]
 1 │ 'a'b
   ·    ┬
   ·    ╰── invalid characters
   ╰────
  help: Try removing them.

❯ 'a b'b
Error: nu::parser::extra_token_after_closing_delimiter

  × Invalid characters after closing delimiter
   ╭─[entry #7:1:6]
 1 │ 'a b'b
   ·      ┬
   ·      ╰── invalid characters
   ╰────
  help: Try removing them.

❯ "a b"c == '"a b"c'
Error: nu::parser::extra_token_after_closing_delimiter

  × Invalid characters after closing delimiter
   ╭─[entry #9:1:6]
 1 │ "a b"c == '"a b"c'
   ·      ┬
   ·      ╰── invalid characters
   ╰────
  help: Try removing them.
```

**Comment by NotTheDr01ds** [comment]
I believe this is a duplicate of #13010 which is closed based on #13131.  I'd recommend going ahead and closing this one as well.

**Comment by fdncred** [comment]
Agreed
